### ISSUE TITLE
Why can't nacos supprt http2 forwarding using nginx

### ISSUE THREAD

**Issue by LiamLea** [alibaba/nacos#10141]
## Issue Description

I wan to figure out why nacos cannot support http2 using nginx?
![2023-03-20_15-52](https://user-images.githubusercontent.com/38973831/226278653-7cc7a313-bd4d-4203-b307-0e9b6b060ff4.png)

**Comment by KomachiSion** [comment]
If using http2, the tcp connection from nginx to server will be closed or reused, which will make your service lost or register frequently.

The problem is nginx's behavior

**Comment by LiamLea** [comment]
> If using http2, the tcp connection from nginx to server will be closed or reused, which will make your service lost or register frequently.
> 
> The problem is nginx's behavior

The Envoy proxy seems to have  the same problem. Is there any http2 proxy realization Nacos supports.

**Comment by KomachiSion** [comment]
No, why want use http2 proxy?  you can only set 9848 port as tcp stream, other port use http2 stream.

**Comment by Maksim-lk** [comment]
> 

If I want to count the status code of gRPC requests to calculate the SLA, 
is there an avaliable realization? 

If Nginx uses TCP model to proxy gRPC requests, it cannot obtain the status code of a single gRPC request

**Comment by LiamLea** [comment]
> No, why want use http2 proxy? you can only set 9848 port as tcp stream, other port use http2 stream.

Thanks. It's ok for me to use tcp forwarding. I just want to figure out what the problem is and it's the flaw of  proxy or nacos. 

I  use istio-proxy(envoy) to forward grpc to nacos and find that the proxy initialize a new http2 connection so that nacos thinks the client is unregistered and disconnect it. I also create an issuer for this in istio community.

**Comment by KomachiSion** [comment]
The reason is that the data (such as registered instances or subscriber) is linked with the connection(client). After the connection is disconnected, nacos think the client has down or shutdown, these datum will be removed as soon as possible to keep the datum consistency.

If the nginx use http2 forward, the connection between nacos might be reuse or be disconnected after one http2 request finished. It will cause the datum problem.

Some `no status` app can use http2 forward, but nacos is a `status` app, so the forward only use tcp.

**Comment by LiamLea** [comment]
Hi @Maksim-lk, I think you can do this at the nacos server which may expose some metrics.
